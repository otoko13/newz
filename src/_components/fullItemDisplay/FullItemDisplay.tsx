import React from 'react';	
import './fullItemDisplay.scss';	
import { INewsItem } from '../newsService';

export interface IFullItemDisplayProps {	
    newsItem?: INewsItem;	
}	

const FullItemDisplay = (props: IFullItemDisplayProps) => {	
    return (
        <div className='FullItemDisplay'>	
            { 
                props.newsItem && 	
                props.newsItem.title	
            }
            { 
                !props.newsItem && 	
                <div className='no-items-message'>Please click on a news item that isn't a link to view it here</div>	
            }	
        </div>	
    );	
}	

export default FullItemDisplay; 