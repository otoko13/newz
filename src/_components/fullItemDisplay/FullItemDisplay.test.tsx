import { act, render } from '@testing-library/react';
import * as React from 'react';
import FullItemDisplay, { IFullItemDisplayProps } from './FullItemDisplay';
import NewsService, { INewsItem } from '../newsService';

jest.mock('../newsService', () => {
    const mockComments: INewsItem[] = [
        { id: 2, descendants: 0, by: 'Maurice', kids: [], time: 1581442689, title: 'Another comment', text: 'I believe', type: 'story' },
        { id: 3, descendants: 0, by: 'Phillip', kids: [], time: 1581442689, title: 'Yet another comment', text: 'I don\'t', type: 'story' },
        { id: 4, descendants: 0, by: 'Kevin', kids: [], time: 1581442689, title: 'One more comment', text: 'Well who the hell are you', type: 'story' },
    ]
    
    return {
        getComments: jest.fn().mockResolvedValue(mockComments),
    }
});

let props: IFullItemDisplayProps;

beforeEach(() => {
    props = {
        newsItem: {
            id: 1,
            descendants: 0,
            by: 'King Julian',
            kids: [2, 3, 4],
            time: 1581442689,
            title: 'A comment',
            text: 'I believe',
            type: 'story',
        },
        hasNext: true,
        hasPrevious: true,
        onNextClick: jest.fn(),
        onPreviousClick: jest.fn(),
    }
});

describe('snapshot', () => {
    it('should look sensible', async () => {
        const { container } = render(<FullItemDisplay {...props} />);
        expect(container.firstChild).toMatchSnapshot();
        await act(async () => {
            await NewsService.getComments({} as INewsItem);
        });
    });
});

describe('no news message', () => {
    it('should not show a please select something message when there is an item', async () => {
        const { queryByText } = render(<FullItemDisplay {...props} newsItem={undefined} />);
        expect(queryByText('Please click on a news item that isn\'t a link to view it here')).not.toBeNull();
        await act(async () => {
            await NewsService.getComments({} as INewsItem);
        });
    });

    it('should show the item instead if there is one', async () => {
        const { queryByText } = render(<FullItemDisplay {...props} />);
        expect(queryByText('Please click on a news item that isn\'t a link to view it here')).toBeNull();
        await act(async () => {
            await NewsService.getComments({} as INewsItem);
        });
    });
});
