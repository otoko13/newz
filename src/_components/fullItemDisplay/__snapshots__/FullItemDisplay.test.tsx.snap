// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot should look sensible 1`] = `
<div
  class="ms-Stack FullItemDisplay css-42"
>
  <div
    class="ms-StackItem css-43"
  >
    <div
      class="ms-Stack NavigationControls css-44"
    >
      <div
        class="ms-StackItem css-43"
      >
        <button
          aria-label="Previous item"
          class="ms-Button ms-Button--icon navigation-button root-45"
          data-is-focusable="true"
          type="button"
        >
          <span
            class="ms-Button-flexContainer flexContainer-46"
            data-automationid="splitbuttonprimary"
          >
            <i
              aria-hidden="true"
              class="ms-Icon root-35 ms-Button-icon icon-48"
              data-icon-name="ChevronLeft"
              role="presentation"
            />
          </span>
        </button>
      </div>
      <div
        class="ms-StackItem css-43"
      >
        <button
          aria-label="Next item"
          class="ms-Button ms-Button--icon navigation-button root-45"
          data-is-focusable="true"
          type="button"
        >
          <span
            class="ms-Button-flexContainer flexContainer-46"
            data-automationid="splitbuttonprimary"
          >
            <i
              aria-hidden="true"
              class="ms-Icon root-35 ms-Button-icon icon-48"
              data-icon-name="ChevronRight"
              role="presentation"
            />
          </span>
        </button>
      </div>
    </div>
  </div>
  <div
    class="ms-StackItem item-content css-53"
  >
    <div
      class="title"
    >
      A comment
    </div>
    <div
      class="by"
    >
      King Julian
    </div>
    <div
      class="date"
    >
      17:38, Tuesday February 11th 2020
    </div>
    <div
      class="content"
    >
      I believe
    </div>
    <div
      class="CommentsSection"
    >
      <div
        class="title"
      >
        Comments:
      </div>
      <div
        class="LoadShimmer"
        style="height: 95px; margin-bottom: 20px;"
        title="loading"
      >
        <div
          class="ms-Shimmer-container top root-54"
        >
          <div
            class="ms-Shimmer-shimmerWrapper shimmerWrapper-55"
            style="width: 100%;"
          >
            <div
              class="ms-Shimmer-shimmerGradient shimmerGradient-56"
            />
            <div
              class="ms-ShimmerElementsGroup-root root-58"
              style="width: auto;"
            >
              <div
                class="ms-ShimmerLine-root root-59"
                style="width: 80%; min-width: auto;"
              >
                <svg
                  class="ms-ShimmerLine-topLeftCorner topLeftCorner-60"
                  height="2"
                  width="2"
                >
                  <path
                    d="M0 2 A 2 2, 0, 0, 1, 2 0 L 0 0 Z"
                  />
                </svg>
                <svg
                  class="ms-ShimmerLine-topRightCorner topRightCorner-61"
                  height="2"
                  width="2"
                >
                  <path
                    d="M0 0 A 2 2, 0, 0, 1, 2 2 L 2 0 Z"
                  />
                </svg>
                <svg
                  class="ms-ShimmerLine-bottomRightCorner bottomRightCorner-62"
                  height="2"
                  width="2"
                >
                  <path
                    d="M2 0 A 2 2, 0, 0, 1, 0 2 L 2 2 Z"
                  />
                </svg>
                <svg
                  class="ms-ShimmerLine-bottomLeftCorner bottomLeftCorner-63"
                  height="2"
                  width="2"
                >
                  <path
                    d="M2 2 A 2 2, 0, 0, 1, 0 0 L 0 2 Z"
                  />
                </svg>
              </div>
            </div>
          </div>
        </div>
        <div
          class="ms-Shimmer-container root-54"
        >
          <div
            class="ms-Shimmer-shimmerWrapper shimmerWrapper-55"
            style="width: 30%;"
          >
            <div
              class="ms-Shimmer-shimmerGradient shimmerGradient-56"
            />
            <div
              class="ms-ShimmerElementsGroup-root root-58"
              style="width: auto;"
            >
              <div
                class="ms-ShimmerLine-root root-64"
                style="width: 100%; min-width: auto;"
              >
                <svg
                  class="ms-ShimmerLine-topLeftCorner topLeftCorner-60"
                  height="2"
                  width="2"
                >
                  <path
                    d="M0 2 A 2 2, 0, 0, 1, 2 0 L 0 0 Z"
                  />
                </svg>
                <svg
                  class="ms-ShimmerLine-topRightCorner topRightCorner-61"
                  height="2"
                  width="2"
                >
                  <path
                    d="M0 0 A 2 2, 0, 0, 1, 2 2 L 2 0 Z"
                  />
                </svg>
                <svg
                  class="ms-ShimmerLine-bottomRightCorner bottomRightCorner-62"
                  height="2"
                  width="2"
                >
                  <path
                    d="M2 0 A 2 2, 0, 0, 1, 0 2 L 2 2 Z"
                  />
                </svg>
                <svg
                  class="ms-ShimmerLine-bottomLeftCorner bottomLeftCorner-63"
                  height="2"
                  width="2"
                >
                  <path
                    d="M2 2 A 2 2, 0, 0, 1, 0 0 L 0 2 Z"
                  />
                </svg>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
